abixen-platform-business-intelligence-service/src/main/java/com/abixen/platform/service/businessintelligence/multivisualisation/service/impl/DatabaseConnectionServiceImpl.java

```diff
@@ -91,16 +92,18 @@ public DatabaseConnection updateDatabaseConnection(DatabaseConnectionForm databa
         log.debug("updateDatabaseConnection() - databaseConnectionForm: " + databaseConnectionForm);
 
         DatabaseConnection databaseConnection = dataSourceConnectionRepository.findOne(databaseConnectionForm.getId());
-        databaseConnection.setName(databaseConnectionForm.getName());
-        databaseConnection.setDatabaseType(databaseConnectionForm.getDatabaseType());
-        databaseConnection.setDatabaseHost(databaseConnectionForm.getDatabaseHost());
-        databaseConnection.setDatabasePort(databaseConnectionForm.getDatabasePort());
-        databaseConnection.setDatabaseName(databaseConnectionForm.getDatabaseName());
-        databaseConnection.setDescription(databaseConnectionForm.getDescription());
-        databaseConnection.setPassword(databaseConnectionForm.getPassword());
-        databaseConnection.setUsername(databaseConnectionForm.getUsername());
-
-        DatabaseConnection updateDatabaseConnection = updateDatabaseConnection(databaseConnection);
+        DatabaseConnection reBuildedDatabaseConnection = databaseConnectionBuilder()
```

Comment:
Rebuild is undesirable, because an entity will have methods like changeDescription(). Check entity i.e. User in ddd-core branch